### PR TITLE
buildkite-agent:  secrecy improvements:  non-store, non-Nix provisioning of secrets

### PR DESCRIPTION
**Pull request by deepfire**
This is the `master` version of https://github.com/NixOS/nixpkgs/pull/31979
###### Motivation for this change

1. Remove the requirement to store secrets in Nix source:
   - SSH keys
   - Buildkite token
2. Make sure that the secrets don't hit the Nix store.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by deepfire** [comment]
cc @domenkozar

**Comment by deepfire** [comment]
Also, part of https://github.com/NixOS/nixpkgs/issues/24288.

File: nixos/modules/services/continuous-integration/buildkite-agent.nix [hunk 1]
```diff
@@ -39,10 +31,12 @@ in
...
       };
 
-      token = mkOption {
-        type = types.str;
+      tokenPath = mkOption {
+        type = types.path;
         description = ''
           The token from your Buildkite "Agents" page.
+
+          A path to the token file.
```
Comment (domenkozar) [inline nixos/modules/services/continuous-integration/buildkite-agent.nix]: Needs clearer explanation that the path is used at runtime, not build time.

File: nixos/modules/services/continuous-integration/buildkite-agent.nix [hunk 2]
```diff
@@ -62,16 +56,20 @@ in
       };
 
       openssh =
-        { privateKey = mkOption {
-            type = types.str;
+        { privateKeyPath = mkOption {
```
Comment (domenkozar) [inline nixos/modules/services/continuous-integration/buildkite-agent.nix]: All three option renames are missing `mkRenamedOptionModule` (grep for examples)

File: nixos/modules/services/continuous-integration/buildkite-agent.nix (hunk 1 above)
Comment (deepfire) [reply nixos/modules/services/continuous-integration/buildkite-agent.nix]: Done.

File: nixos/modules/services/continuous-integration/buildkite-agent.nix (hunk 2 above)
Comment (deepfire) [reply nixos/modules/services/continuous-integration/buildkite-agent.nix]: Done.

**Review by zimbatm — APPROVED** [review]
nice!

**Comment by deepfire** [comment]
@domenkozar, anything else to move this forward?

**Pull request merged**